subsys/zigbee/osif/zb_nrf_transceiver.c

```diff
@@ -151,63 +166,95 @@ void zb_trans_get_rssi(zb_uint8_t *rssi_value_p)
 void zb_trans_set_channel(zb_uint8_t channel_number)
 {
 	LOG_DBG("Function: %s, channel number: %d", __func__, channel_number);
-	nrf_802154_channel_set(channel_number);
+
+	radio_api->set_channel(radio_dev, channel_number);
 }
 
 /* Sets the transmit power. */
 void zb_trans_set_tx_power(zb_int8_t power)
 {
 	LOG_DBG("Function: %s, power: %d", __func__, power);
-	nrf_802154_tx_power_set(power);
+
+	radio_api->set_txpower(radio_dev, power);
+	state_cache.power = power;
 }
 
 /* Gets the currently set transmit power. */
 void zb_trans_get_tx_power(zb_int8_t *power)
 {
-	*power = (zb_int8_t)nrf_802154_tx_power_get();
+	__ASSERT_NO_MSG(state_cache.power != SCHAR_MIN);
+
+	*power = state_cache.power;
+
 	LOG_DBG("Function: %s, power: %d", __func__, *power);
 }
 
 /* Configures the device as the PAN coordinator. */
 void zb_trans_set_pan_coord(zb_bool_t enabled)
 {
+	struct ieee802154_config config = { .pan_coordinator = enabled };
+
 	LOG_DBG("Function: %s, enabled: %d", __func__, enabled);
-	nrf_802154_pan_coord_set((bool)enabled);
+
+	radio_api->configure(radio_dev,
+			     IEEE802154_CONFIG_PAN_COORDINATOR,
+			     &config);
 }
 
 /* Enables or disables the automatic acknowledgments (auto ACK) */
 void zb_trans_set_auto_ack(zb_bool_t enabled)
 {
+	struct ieee802154_config config = {
+		.auto_ack_fpb = {
+			.enabled = enabled,
+			.mode = IEEE802154_FPB_ADDR_MATCH_ZIGBEE
+		}
+	};
+
 	LOG_DBG("Function: %s, enabled: %d", __func__, enabled);
-	nrf_802154_auto_ack_set((bool)enabled);
+
+	radio_api->configure(radio_dev,
+			     IEEE802154_CONFIG_AUTO_ACK_FPB,
+			     &config);
 }
 
 /* Enables or disables the promiscuous radio mode. */
 void zb_trans_set_promiscuous_mode(zb_bool_t enabled)
 {
+	struct ieee802154_config config = {
+		.promiscuous = enabled
+	};
+
 	LOG_DBG("Function: %s, enabled: %d", __func__, enabled);
-	nrf_802154_promiscuous_set((bool)enabled);
+
+	radio_api->configure(radio_dev,
+			     IEEE802154_CONFIG_PROMISCUOUS,
+			     &config);
 }
 
 /* Changes the radio state to receive. */
 void zb_trans_enter_receive(void)
 {
 	LOG_DBG("Function: %s", __func__);
-	(void)nrf_802154_receive();
+
+	radio_api->start(radio_dev);
```

Comment:
Jest checking: is it correct that if `radio_api->start` is called during transmission, it will wait until the transmission is finished, and center the RECEIVE state afterwards?
If not, than we have some state conflicts in TX/RX functions.

Reply:
`radio_api->start(radio_dev);` calls `nrf_802154_receive()` internally. I'm not sure if `nrf_802154_receive()` waits for completion of transmit and enters RECEIVE afterwards. Why do we need this? I mean where a state conflict can happen?

Reply:
If `nrf_802154_receive()` is called during transmission, it aborts the transmission and enters RX state.

Reply:
There was a chance that the `start` operation will simply fail if called during transmission. In this case the state would be updated to RX, but the radio would be in TX.